floorplan: replace surprises and documentation with actionable error messages

oharboe opened the pull request:
I've been suckerpunched several times by CORE_UTILIZATION overriding DIE/CORE_AREA.

Remove documentation/rules and replace it with checks and actionable error messages.

A floorplan config suckerpunch can manifest itself as late as in macro placement with an [inactionable error message](https://github.com/The-OpenROAD-Project/OpenROAD/pull/5735)....

oharboe committed:
floorplan: replace surprises and documentation with actionable error messages

I've been surprised several times by CORE_UTILIZATION overriding
DIE/CORE_AREA.

Remove documentation/rules and replace it with checks and
actionable error messages.

Signed-off-by: Øyvind Harboe <oyvind.harboe@zylin.com>

oharboe committed:
Merge remote-tracking branch 'origin/master' into HEAD

oharboe committed:
floorplan: mutually exclusive floorplan methods, pick one

Signed-off-by: Øyvind Harboe <oyvind.harboe@zylin.com>

oharboe committed:
Merge remote-tracking branch 'origin/master' into HEAD

maliberty commented on docs/user/FlowVariables.md:
```diff
@@ -265,16 +265,18 @@ configuration file.
 | `MACRO_HALO_X`         | Set macro halo for x-direction. Only available for ASAP7 PDK. |
 | `MACRO_HALO_Y`         | Set macro halo for y-direction. Only available for ASAP7 PDK. |
 
+The various methods to specify the die and core area(`FLOORPLAN_DEF`, `FOOTPRINT`, `DIE_AREA`, `CORE_AREA` and `CORE_UTILIZATION`) are mutually exclusive. If two methods are specified, floorplan.tcl will exit with an error requiring that a single method is used.
```
There are four methods
* FLOORPLAN_DEF
* DIE_AREA and CORE_AREA
* FOOTPRINT  (obsolete and should be undocumented)
* CORE_UTILIZATION with optional CORE_ASPECT_RATIO

Its not right to say DIE_AREA and CORE_AREA are mutually exclusive.

oharboe replied:
Well spotted. Fixed.

oharboe committed:
doc: DIE_AREA is shorthand for a method to specify floorplan

There is an error message if CORE_AREA is not specified together
with DIE_AREA, but it is close enough to the truth to say that
DIE_AREA is a method to specify floorplan.

Signed-off-by: Øyvind Harboe <oyvind.harboe@zylin.com>
